server/src/main/java/org/elasticsearch/plugins/Plugin.java

```diff
@@ -84,7 +143,10 @@ public abstract class Plugin implements Closeable {
      * @param telemetryProvider           An interface for distributed tracing
      * @param allocationService           A service to manage shard allocation in the cluster
      * @param indicesService              A service to manage indices in the cluster
+     *
+     * @deprecated New services will only be added to {@link PluginServices}
      */
+    @Deprecated
```

Comment:
This can just be left around for compatibility, but any new services would be added to `PluginServices` only

Reply:
We don't need to do this, or at least we can remove this once serverless is updated. There are no compatibility guarantees for non-stable plugins.

Reply:
I'll migrate all the existing implementations as a separate PR